datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/common/mappers/FineGrainedLineagesMapper.java

```diff
@@ -0,0 +1,51 @@
+package com.linkedin.datahub.graphql.types.common.mappers;
+
+import com.linkedin.common.urn.Urn;
+import com.linkedin.datahub.graphql.generated.FineGrainedLineage;
+import com.linkedin.datahub.graphql.generated.SchemaFieldRef;
+import com.linkedin.dataset.FineGrainedLineageArray;
+
+import javax.annotation.Nonnull;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class FineGrainedLineagesMapper {
+
+  public static final FineGrainedLineagesMapper INSTANCE = new FineGrainedLineagesMapper();
+
+  public static List<FineGrainedLineage> map(@Nonnull final FineGrainedLineageArray fineGrainedLineages) {
+    return INSTANCE.apply(fineGrainedLineages);
+  }
+
+  public List<com.linkedin.datahub.graphql.generated.FineGrainedLineage> apply(@Nonnull final FineGrainedLineageArray fineGrainedLineages) {
+    final List<com.linkedin.datahub.graphql.generated.FineGrainedLineage> result = new ArrayList<>();
+    if (fineGrainedLineages.size() == 0) {
+      return result;
+    }
+
+    for (com.linkedin.dataset.FineGrainedLineage fineGrainedLineage : fineGrainedLineages) {
+      com.linkedin.datahub.graphql.generated.FineGrainedLineage resultEntry = new com.linkedin.datahub.graphql.generated.FineGrainedLineage();
+      if (fineGrainedLineage.hasUpstreams()) {
+        resultEntry.setUpstreams(fineGrainedLineage.getUpstreams().stream()
+            .filter(entry -> entry.getEntityType().equals("schemaField"))
```

Comment:
nitpick: do we have a constant for SCHEMA_FIELD_ENTITY_NAME?

Reply:
why yes we do good call!